system-test/test.sh

```diff
@@ -38,23 +38,42 @@ npm pack >/dev/null
 VERSION=$(node -e "console.log(require('./package.json').version);")
 PROFILER="$PWD/pprof-$VERSION.tgz"
 
+if [[ "$VERIFY_TIME_LINE_NUMBERS" == "true" ]]; then
+  BENCHDIR="$PWD/system-test/busybench-js"
+  BENCHPATH="src/busybench.js"
+else
+  BENCHDIR="$PWD/system-test/busybench"
+  BENCHPATH="build/src/busybench.js"
+fi
+
 TESTDIR=$(mktemp -d)
-cp -r "$PWD/system-test/busybench" "$TESTDIR"
+cp -r "$BENCHDIR" "$TESTDIR/busybench"
 cd "$TESTDIR/busybench"
 
 retry npm_install pify @types/pify typescript gts @types/node >/dev/null
 retry npm_install --nodedir="$NODEDIR" \
-    ${BINARY_HOST:+--pprof_binary_host_mirror=$BINARY_HOST} \
+    $([ -z "$BINARY_HOST" ] && echo "--build-from-source=pprof" \
+        || echo "--pprof_binary_host_mirror=$BINARY_HOST")\
     "$PROFILER">/dev/null
 
-npm run compile >/dev/null
+if [[ "$VERIFY_TIME_LINE_NUMBERS" != "true" ]]; then
+  npm run compile
+fi
 
 node -v
-node --trace-warnings build/src/busybench.js 10
-ls -l
+node --trace-warnings "$BENCHPATH" 10 $VERIFY_TIME_LINE_NUMBERS
+
+if [[ "$VERIFY_TIME_LINE_NUMBERS" == "true" ]]; then
+  pprof -lines -top -nodecount=2 time.pb.gz | \
+      grep "busyLoop.*src/busybench.js:33"
```

Comment:
I would change ":33" to ":34" here once and make sure the test fails, just to double-check that the "verify time line numbers" flag machinery works correctly. As if it doesn't the test would happily pass without the line numbers feature working.

Reply:
Done